### DISCUSSION TITLE
Track total on-chain transaction cost

### DISCUSSION THREAD

**Discussion by bonomat** [lightningdevkit/rust-lightning#2405] [Ideas / Feature Requests]
We would like to be able to track the total on-chain cost associated with a channel, which means: 

- funding transaction fee: if the channel was opened by us
- commitment transaction fee: if we have paid it
- claimable output transaction fee

What is the best way to track this?

**Comment by wpaulino** [comment]
> funding transaction fee: if the channel was opened by us

You should be able to track this one already since you're funding the channel, just take note of the transaction fee prior to handing it off to `ChannelManager::funding_transaction_generated`.

> commitment transaction fee: if we have paid it

This one will require changes to LDK. We can probably just extend `ChannelClosed` to include the closing transaction, and along with the capacity being added in https://github.com/lightningdevkit/rust-lightning/pull/2387, you can figure out the fee by subtracting the total output amount from the capacity.

> claimable output transaction fee

This one is a bit more involved. I assume you're referring to `SpendableOutput` claims? You should be able to track those as well as you're funding those transactions. The issue here comes from the HTLC transactions, we'd need to expose those somehow, and they generally happen after already receiving the `ChannelClosed` event.

**Reply by TheBlueMatt** [reply]
> We can probably just extend ChannelClosed to include the closing transaction

Sadly we don't know for sure which commitment transaction will be confirmed when we generate the `ChannelClosed` event (see-also discussion at https://github.com/orgs/lightningdevkit/discussions/2407).

In general, doing this may be simplest done by simply tracking the transactions themselves - when a channel is live/funded, store the outpoint for the channel, and from there just look at transactions spending down the tree and add them up. We should have better logic for this in LDK via the `ChannelMonitor`, but currently dont.

**Comment by TheBlueMatt** [comment]
Opened  #2494 to track a better way to do this, without making downstream-from-LDK code calculate this manually (though that should also be pretty doable).
